server/service/src/stocktake/update.rs

```diff
@@ -302,9 +302,11 @@ fn generate_stock_in_out_or_update(
 
     let update_inventory_adjustment_reason = generate_update_inventory_adjustment_reason(
         invoice_line_id.clone(),
-        row.inventory_adjustment_reason_id,
+        row.inventory_adjustment_reason_id.clone(),
     );
 
+    log_stock_changes(ctx, stock_line_row.clone(), row.clone())?;
```

Comment:
This needs to move up, above the check for `delta == 0.0`

Currently only saves logs if there is an inventory adjustment, but should work if counted # packs is the same 😁

Reply:
[bceb341](https://github.com/msupply-foundation/open-msupply/pull/4328/commits/bceb341509cb37783750bca27fe060443c719b46)